### PR TITLE
feature/CVSB-178

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

File: src/pages/testing/test-creation/test-types-list/test-types-list.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+page-tests-types-list {
+  //.item {
```
Comment (acozianu) [inline src/pages/testing/test-creation/test-types-list/test-types-list.scss]: Please remove the comments

File: src/pages/testing/test-creation/test-types-list/test-types-list.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  .item-ios.list-item-height {
+    min-height: $list-ios-item-height;
+
+    h3 {
+      font-size: 17px;
+    }
+  }
+}
+
+//ion-app.ios page-tests-list [padding] .scroll-content {
```
Comment (acozianu) [inline src/pages/testing/test-creation/test-types-list/test-types-list.scss]: Also here

File: src/providers/global/http.service.ts [hunk 3]
```diff
@@ -26,8 +28,9 @@ export class HTTPService {
     return this.http.get<DefectCategoryModel[]>(`${AppConfig.API_URL}${API.GET_DEFECTS}`)
   }
 
-  getTestTypes(): Observable<TestTypeModel[]> {
-    return this.http.get<TestTypeModel[]>(`${AppConfig.API_URL}${API.GET_TESTTYPES}`);
+  getTestTypes(): Observable<TestTypesModel[]> {
+    // return this.http.get<TestTypesModel[]>(`${AppConfig.API_URL}${API.GET_TESTTYPES}`);
```
Comment (acozianu) [inline src/providers/global/http.service.ts]: Please fix

File: src/pages/testing/test-creation/test-types-list/test-types-list.scss (hunk 2 above)
Comment (danieliordache) [reply src/pages/testing/test-creation/test-types-list/test-types-list.scss]: done

File: src/pages/testing/test-creation/test-types-list/test-types-list.scss (hunk 1 above)
Comment (danieliordache) [reply src/pages/testing/test-creation/test-types-list/test-types-list.scss]: done

File: src/providers/global/http.service.ts (hunk 3 above)
Comment (danieliordache) [reply src/providers/global/http.service.ts]: done

File: package.json [hunk 4]
```diff
@@ -56,16 +56,18 @@
     "cordova-plugin-file": "^5.0.0",
     "cordova-plugin-inappbrowser": "^2.0.1",
     "cordova-plugin-ionic-webview": "^1.2.1",
+    "cordova-plugin-network-information": "git+https://github.com/apache/cordova-plugin-network-information.git",
```
Comment (adpetrescu) [inline package.json]: where do you need this?

File: src/models/reference-data-models/test-types.model.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export interface TestTypesModel {
+  id: string;
+  name: string;
+  forVehicleType: string[];
+  forVehicleSize: string[];
+  forVehicleConfiguration: string[];
+  forVehicleAxles: number[];
+  nextTestTypesOrCategories?: TestTypesModel[] | null;
```
Comment (adpetrescu) [inline src/models/reference-data-models/test-types.model.ts]: why do you need also `null`?

File: src/pages/testing/defects/add-defect/add-defect.html [hunk 6]
```diff
@@ -15,33 +15,40 @@
...
+                class="list-item-height"
+                *ngFor="let deficiency of filteredDeficiencies | byVehicleType: vehicleType"
+                (click)="selectDeficiency(deficiency)">
       <span>
           <h3>
             {{deficiency.ref}}
             <ion-badge item-end text-uppercase
                        [color]="defectsService.getBadgeColor(deficiency.deficiencyCategory)"
-                       [ngClass]="{'badge__text-black': deficiency.deficiencyCategory === 'Minor'}">
+                       [ngClass]="{'badge-text-black': deficiency.deficiencyCategory === 'Minor'}">
```
Comment (adpetrescu) [inline src/pages/testing/defects/add-defect/add-defect.html]: can you please use an enum for `Minor`?

File: src/pages/testing/test-creation/complete-test/complete-test.spec.ts [hunk 7]
```diff
@@ -21,7 +21,7 @@ describe('Component: CompleteTestPage', () => {
   let alertCtrl: AlertController;
   let storageServiceSpy: any;
 
-  const defects: DefectCategoryModel[] =  DefectsDataMock.DefectsData
+  const defects: DefectCategoryModel[] = DefectsDataMock.DefectsData
   const addedDefect: DefectDetailsModel = {
     ref: '1.1.a',
     deficiencyCategory: 'Major',
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/complete-test/complete-test.spec.ts]: enum for Major?

File: src/providers/vehicle.service.ts [hunk 8]
```diff
@@ -30,10 +30,10 @@ let historyTest3 = new VehicleTestModel("Annual test", true, new Date(2016, 5, 1
...
 
-let vehicle1 = new VehicleModel("AA12 BCD", "123ADF213DAS", "psv", 3, "IVECO", "1S34RS", 12354, [historyTest3, historyTest4, historyTest5]);
-let vehicle2 = new VehicleModel("AB34 ERT", "123ADF213DAS", "Trailer", 2, "DENNISON", "REFRIGERATOR", 45125, [historyTest1, historyTest2]);
+let vehicle1 = new VehicleModel("AA12 BCD", "123ADF213DAS", "psv", 2, "IVECO", "1S34RS", 12354, 'rigid', 'small', [historyTest3, historyTest4, historyTest5]);
+let vehicle2 = new VehicleModel("AB34 ERT", "123ADF213DAS", "trailer", 2, "DENNISON", "REFRIGERATOR", 45125,'articulated', 'large',[historyTest1, historyTest2]);
 
 export const VEHICLES: VehicleModel[] = [
   vehicle1,
-  vehicle2
+  // vehicle2
```
Comment (adpetrescu) [inline src/providers/vehicle.service.ts]: can you readd this?

File: package.json (hunk 4 above)
Comment (danieliordache) [reply package.json]: done

File: src/models/reference-data-models/test-types.model.ts (hunk 5 above)
Comment (danieliordache) [reply src/models/reference-data-models/test-types.model.ts]: done

File: src/pages/testing/defects/add-defect/add-defect.html (hunk 6 above)
Comment (danieliordache) [reply src/pages/testing/defects/add-defect/add-defect.html]: done

File: src/providers/vehicle.service.ts (hunk 8 above)
Comment (danieliordache) [reply src/providers/vehicle.service.ts]: done

Comment (danieliordache) [reply src/providers/vehicle.service.ts]: done

**Pull request merged**